Tools/SwiftBrowser/Source/SwiftBrowser.swift

```diff
@@ -32,11 +32,28 @@ struct SwiftBrowserApp: App {
 
     @State private var mostRecentURL: URL? = nil
 
+    func addProtocolIfNecessary(_ address: String) -> String {
+        if address.contains("://") || address.hasPrefix("data:") || address.hasPrefix("about:") {
+            return address
+        }
+        return "http://\(address)"
+    }
+
     var body: some Scene {
         WindowGroup(for: CodableURLRequest.self) { $request in
             BrowserView(url: $mostRecentURL, initialRequest: request.value)
         } defaultValue: {
-            let url = URL(string: homepage)!
+            var parsedURL: URL? = nil
+            let args = CommandLine.arguments
```

Comment:
this can be factored out into a `CommandLine+Extras.swift` file and then have a function like

```
extension CommandLine {
    static func value(for flag: String) -> String? {
        guard let flagIndex = arguments.firstIndex(of: flag) else {
            return nil
        }

        let valueIndex = arguments.index(after: flagIndex)
        guard valueIndex < arguments.endIndex else {
            return nil
        }

        return arguments[valueIndex]
    }
}
```